### PR TITLE
Add doc for impl From for Std Error

### PR DESCRIPTION
**Pull request by phungleson**
As part of issue #51430 (cc @skade).

I am not sure if it is going to a correct direction so put up here so that people can comment.

### CODE REVIEW

**Comment by pietroalbini** [comment]
Ping from triage @skade! This PR needs your review.

**Comment by TimNN** [comment]
Ping from triage @skade / @rust-lang/docs: This PR requires your review.

File: src/libstd/error.rs [hunk 1]
```diff
@@ -152,20 +152,27 @@ pub trait Error: Debug + Display {
 
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<'a, E: Error + 'a> From<E> for Box<dyn Error + 'a> {
+    /// Converts a type of [`Error`] into a box of dyn [`Error`]
```
Comment (GuillaumeGomez) [inline src/libstd/error.rs]: Please add a dot at the end of the sentence.

**Comment by GuillaumeGomez** [comment]
Also, this isn't only a doc PR, you added `#[inline]`. Please add missing examples.

cc @rust-lang/compiler

**Comment by phungleson** [comment]
I have fixed the line and remove `inline` to make the PR simple.

**Comment by GuillaumeGomez** [comment]
Still missing examples (also you remove the `[inline]` adds so no need for @rust-lang/compiler team anymore).

**Comment by phungleson** [comment]
Hey, I am currently traveling, will look at this again in around 3 weeks.

File: src/libstd/error.rs [hunk 2]
```diff
@@ -152,20 +152,54 @@ pub trait Error: Debug + Display {
...
+    ///     }
+    /// }
+    ///
+    /// fn main() {
+    ///     let an_error = AnError;
+    ///     assert!(0 == mem::size_of_val(&an_error));
+    ///     let a_boxed_error = Box::<Error>::from(an_error);
+    ///     assert!(mem::size_of::<Box<dyn Error>>() == mem::size_of_val(&a_boxed_error))
+    /// }
+    /// ```
```
Comment (phungleson) [inline src/libstd/error.rs]: Hey @GuillaumeGomez I have added a simple example, does it look ok?

If yes I will do similar things to other places.

Comment (GuillaumeGomez) [reply src/libstd/error.rs]: Looks good yes!

**Comment by GuillaumeGomez** [comment]
Please remove all `fn main()` functions, just keep the code inside. `rustdoc`  wraps the code in a `fn main` in any case (except for `extern crate` stuff) if there is none so in here it's just noise.

**Comment by phungleson** [comment]
Cool thanks @GuillaumeGomez `main` has been removed

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Pull request merged**